sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -396,13 +719,15 @@ class GpuHashAggregateIterator(
    * @return lazy spillable batch which has NOT been marked spillable
    */
   private def concatenateAndMerge(
-      batches: mutable.ArrayBuffer[LazySpillableColumnarBatch]): LazySpillableColumnarBatch = {
+      batches: mutable.ArrayBuffer[SpillableColumnarBatch]): SpillableColumnarBatch = {
+    // TODO: concatenateAndMerge (and calling code) could output a sequence
```

Comment:
What is this TODO for? It looks like it is already handled.

Reply:
I should turn it into an issue. Right the code insists on outputting a single batch whether in partial or final aggregation modes. The todo is for the partial aggregation case, where it should be OK to output multiple smaller batches, letting the shuffle handle things later. OK to file this as a follow on?

Reply:
https://github.com/NVIDIA/spark-rapids/issues/7817